src/Compilers/CSharp/Test/Semantic/Semantics/InterpolationTests.cs

```diff
@@ -10316,6 +10316,32 @@ public void DisallowedInExpressionTrees()
             );
         }
 
+        [Fact, WorkItem(55114, "https://github.com/dotnet/roslyn/issues/55114")]
+        public void AsStringInExpressionTrees_01()
+        {
+            var code = @"
+using System;
+using System.Linq.Expressions;
+
+Expression<Func<string, string>> e = o => $""{o.Length}"";";
```

Comment:
The code suggests that there's also a non-generic `Expression` scenario (see `IsGenericOrNonGenericExpressionType`). Do we need to cover that as well?